src/coreclr/src/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/FieldFixupSignature.cs

```diff
@@ -14,6 +14,7 @@ namespace ILCompiler.DependencyAnalysis.ReadyToRun
 {
     public class FieldFixupSignature : Signature
     {
+        public const int MaxCheckableOffset = 0x1FFFFFFF;
```

Comment:
What was the issue with larger offsets?  Are they now unchecked?

Reply:
They are now unchecked. The problem is that the encoding uses the compressed uint encoding from metadata which cannot encode a number greater than 0x1FFFFFFF.

In practice, the need to do this is so rare, that it isn't worth changing the encoding.

Reply:
Shouldn't we fail compilation for large offsets then?  I see throwing a `RequiresRuntimeJitException` in one `if` branch, but not in others.